storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/dataaccess/V4FinalizedRocksDbDao.java

```diff
@@ -76,7 +80,9 @@ public void addFinalizedBlock(final SignedBeaconBlock block) {
 
     @Override
     public void addFinalizedState(final Bytes32 blockRoot, final BeaconState state) {
-      transaction.put(V4SchemaFinalized.FINALIZED_STATES_BY_ROOT, blockRoot, state);
+      if (state.getSlot().mod(stateStorageFrequency).equals(UnsignedLong.ZERO)) {
```

Comment:
The risk with this is that if we happen to have an empty slot where the state would have been stored we'd wind up with a significantly bigger gap than we expected.  It would be easy to have skipped slots at a few archive points in a row creating a huge gap.

We could probably start with this but I think we need to be smarter and effectively store the state in effect at our archive slot even if it didn't have a block.